cpp/src/io/parquet/page_decode.cuh

```diff
@@ -691,6 +696,10 @@ __device__ void gpuUpdateValidityOffsetsAndRowIndices(int32_t target_input_value
 
   PageNestingDecodeInfo* nesting_info_base = s->nesting_info;
 
+  // need this to ensure input_value_count is read by all threads before s->input_value_count
+  // is modified below (just in case input_value count >= target_input_value_count).
+  __syncwarp();
```

Comment:
This one I'm not so sure about needing. In the worst case, thead 0 sets the local var, skips the loop (and the syncwarp within it) and then overwrites the shared value before other threads read it. But in that case it will just overwrite with the same value.

Reply:
do we actually need to update s->nz_count, s->input_value_count and s->input_row_count if we never enter the loop?

Reply:
I'm thinking no...they shouldn't have changed if the loop wasn't entered. But I'll admit this is one of the parts of the parquet code that I understand the least.

Reply:
If that's the case, we should be able to return early if initially `input_value_count >= target_input_value_count`, right?
That would simplify the logic and prevent the tool from reporting the race condition.
CC @nvdbaranec

Reply:
I made the change and verified that racecheck is happy